dsl_parser/tasks.py

```diff
@@ -155,6 +155,100 @@ def _find_idd_creating_functions(plan) -> List[functions.IDDSpec]:
     return handler.dependencies
 
 
+def _find_rel_target_by_iddspec(relationships, idd_spec):
+    """Find the target instance that matches the relationship in IDDSpec
+
+    The node-instance can have many relationships, so we need to find one
+    that matches the target node name & type.
+    If there's multiple relationships of the same type to the same node,
+    then we will just select the first instance, because we have no way
+    to distinguish them.
+    (that case shouldn't be possible anyway)
+    """
+    assert idd_spec.scope == 'node_template_relationship'
```

Comment:
Can we either convert this to a check that won't disappear with -O, or just drop it if we don't care whether it's executed?

Reply:
no, this is for the human reading this (or writing this)
I prefer to write it like this, rather than in the docstring

Reply:
OK, always strikes me as pointless but fine.